bindings/imap-codec-python/examples/common/__init__.py

```diff
@@ -0,0 +1,42 @@
+import sys
+from enum import Enum
+from typing import Optional
+
+COLOR_SERVER = "\x1b[34m"
+COLOR_CLIENT = "\x1b[31m"
+RESET = "\x1b[0m"
+
+
+class Role(Enum):
+    Client = 1
+    Server = 2
+
+
+def read_more(buffer: bytearray, role: Role):
+    if not buffer:
```

Comment:
If `buffer` is `None`, the caller would run into an Exception in line 28:
```
> python -c "from examples.common import read_more; read_more(None, None)"      
S: nothing

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "...\examples\common\__init__.py", line 28, in read_more
    buffer += line.encode()
TypeError: unsupported operand type(s) for +=: 'NoneType' and 'bytes'
```

But, why stop at `None`? The parameters could actually be anything:
```
> python -c "from examples.common import read_more; read_more('hello', 'world')"
.. i like strings

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "...\examples\common\__init__.py", line 28, in read_more
    buffer += line.encode()
TypeError: can only concatenate str (not "bytes") to str
```

(Yes, the prompt is different in the two cases.)

That's all fine here, because the caller is violating the contract of the function and deserves an exception for that. Also, nothing bad has happend. (We should definetily prevent accidental misuse, like treating `None` as a wildcard to delete all entries in the database or something like that.)
Even then, the only other thing we could do here would be adding a lot of assertion code at the start of functions and then ... well ... raise an exception.